internal/services/dns/dns_zone_resource.go

```diff
@@ -374,3 +383,16 @@ func flattenArmDNSZoneSOARecord(input *recordsets.RecordSet) []interface{} {
 
 	return output
 }
+
+func soaRecordHostName(ctx context.Context, timeout time.Duration, recordSetsClient *recordsets.RecordSetsClient, id zones.DnsZoneId) (*string, error) {
+	ctx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+
+	soaRecord := recordsets.NewRecordTypeID(id.SubscriptionId, id.ResourceGroupName, id.DnsZoneName, recordsets.RecordTypeSOA, "@")
+	soaRecordResp, err := recordSetsClient.Get(ctx, soaRecord)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving %s: %+v", id, err)
+	}
+
+	return soaRecordResp.Model.Properties.SOARecord.Host, nil
```

Comment:
Here wants nil check on the chained properties that is a pointer, if it is nil, you can return an error.

Reply:
ok, I will try. thanks